src/testRunner/unittests/evaluation/superInStaticInitializer.ts

```diff
@@ -0,0 +1,147 @@
+describe("unittests:: evaluation:: superInStaticInitializer", () => {
+    it("super-property-get in es2015", () => {
+        const result = evaluator.evaluateTypeScript(`
+            export function main() {
+                class Base {
+                    static x = 1;
+                }
+                class Derived extends Base {
+                    static y = super.x;
+                }
+                return [
+                    Base,
+                    Derived
+                ];
+            }
+        `, { target: ts.ScriptTarget.ES2015 });
+        const [Base, Derived] = result.main();
+        assert.strictEqual(Base.x, 1);
+        assert.strictEqual(Derived.y, 1);
+    });
+    it("super-property-set in es2015", () => {
+        const result = evaluator.evaluateTypeScript(`
+            export function main() {
+                class Base {
+                    static x = 1;
+                }
+                class Derived extends Base {
+                    static y = super.x++;
+                }
+                return [
+                    Base,
+                    Derived
+                ];
+            }
+        `, { target: ts.ScriptTarget.ES2015 });
+        const [Base, Derived] = result.main();
+        assert.strictEqual(Base.x, 1);
+        assert.strictEqual(Derived.x, 2);

```

Comment:
... `super.x++` increments `this.x`, but not `Base.x`? OK, I guess.

Reply:
Yeah, that's how it works :)

`++` is two operations, a `[[Get]]` and a `[[Set]]`. The `[[Get]]` walks the prototype chain to read, the `[[Set]]` only walks the prototype chain to look for a setter. If it finds the setter, it calls it with `this` as the receiver. If there's no setter, it sets the property on `this` (not the base class).

Reply:
That oddity is one of the reasons we don't downlevel `SuperProperty` in ES5 unless its a call, the helper necessary to emulate `Reflect.get` and `Reflect.set` isn't worth the overhead.